hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -157,14 +157,22 @@ public void accept(MessageSendingResult result) {
                 rateLimiter.registerSuccessfulSending();
                 handleMessageSendingSuccess(message, result);
             } else {
-                handleFailedSending(message, result);
-                message.incrementRetryCounter();
+                if (result.isUnauthorized()) {
+                    message.incrementNumberOfUnauthorizedRequests();
+                }
```

Comment:
1) move this stuff to `handleFailedSending`
2) don't send messages from the same thread that handles responses